aten/src/ATen/cuda/tunable/TunableOp.h

```diff
@@ -21,6 +21,26 @@
 #include <string>
 #include <unordered_map>
 #include <vector>
+#include <deque>
+
+class FixedSizeStack {
+  private:
+      std::deque<std::string> stack;
+      const size_t max_size;
+
+  public:
+      FixedSizeStack(size_t size) : max_size(size) {}
+
+      void push(const std::string& value) {
+          if (stack.size() >= max_size) {
+              stack.pop_front(); // Remove the oldest entry
+          }
+          stack.push_back(value); // Add new entry
+      }
+
+      auto rbegin() { return stack.rbegin(); }
+      auto rend() { return stack.rend(); }
+};
 
```

Comment:
Please move inside namespace at::cuda::tunable and also put inside anonymous namespace.  We don't need to expose this FixedSizeStack publicly.